### DISCUSSION TITLE
How to set service.name and have it show up in Jaeger?

### DISCUSSION THREAD

**Discussion by ghost** [open-telemetry/opentelemetry-cpp#1125] [Q&A] [answered]
I have read several discussions that would seem to be related to this question but I am not sure what the trick is to setting service.name. I am assuming that this is used in the Jaeger UI where I am seeing "unknown_service" for everything I do.

**Comment by wxl374** [comment, marked as answer]
sdk::resource::ResourceAttributes attributes = {{"service.name", "my_service_name"}};
    auto resource = sdk::resource::Resource::Create(attributes);
    auto provider = nostd::shared_ptr<trace::TracerProvider>(new sdk::trace::TracerProvider(std::move(processor), resource));
